tests/net/socket/tcp/src/main.c

```diff
@@ -234,6 +235,114 @@ void test_v6_send_recv(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
+/* Test the stack behavior with a resonable sized block data, be sure to have multiple packets */
+#define TEST_BLOCK_SIZE 60000
+
+static uint8_t testBlockTcp[TEST_BLOCK_SIZE];
+static uint8_t testBlockTcpRx[TEST_BLOCK_SIZE + 10];
+
+#define TCP_SERVER_STACK_SIZE 4096
```

Comment:
This and the TEST_BLOCK_SIZE will most definitely mean that the unit test cannot be run on all the devices it is currently running by twister. I just wonder could we make these numbers a bit smaller?

Reply:
I'll give it a try, the challenge that is needs to be a certain size to trigger the issue...

Reply:
Yes, but you can just keep sending the same buffer over and over, the data doesn't need to be unique (of course it would still be good to verify data integrity after each iteration).

Reply:
I have significantly reduced this, the only thing left is 2048 bytes of stack size, the issues still reproduces without the fixes